### PR TITLE
Support G- strkey public keys as arguments

### LINKED ISSUE #53: CLI: Map strkeys (e.g. G...) to FixedBinary<32>
People will naturally want to use Stellar's strkey format for keys as inputs to functions or arguments to things. We should map this to `FixedBinary<32>` or to some other type, so that people can continue to use those keys. ~Possibly now is the right time to finish implementing the `PublicKey` type that is still in the XDR.~

cc @paulbellamy @jayz22 @jonjove 

I'm revisiting this because of this question in Discord by @fsodano:
https://discord.com/channels/897514728459468821/966788672164855829/1003810451408306246

### PR DESCRIPTION
**Pull request by 2opremio**
Closes #53

### CODE REVIEW

File: src/strval.rs [hunk 1]
```diff
@@ -61,12 +63,24 @@ pub fn from_string(s: &str, t: &ScSpecTypeDef) -> Result<ScVal, StrValError> {
...
-                        .try_into()
-                        .map_err(|_| StrValError::InvalidValue)?,
-                ))),
+                _ =>
+                // it could be a G- strkey
+                {
+                    match StrkeyPublicKeyEd25519::from_string(s) {
+                        Ok(key) => {
+                            let b = &key.0;
+                            ScVal::Object(Some(ScObject::Binary(
```
Comment (paulbellamy) [inline src/strval.rs:72]: Shouldn't this be an ScPublicKey? Or maybe that type is missing from the ScSpecTypeDef?

Comment (2opremio) [reply src/strval.rs:72]: I am using `stellar_strkey::StrkeyPublicKeyEd25519` I can't find `ScPublicKey`.

Comment (leighmcculloch) [reply src/strval.rs:72]: ScPublicKey and ScHash aren't being used anywhere at the moment, even within the host. All the examples that do have keys use `Binary`.

**Review by leighmcculloch — APPROVED** [review]
Nice 🎉 Thanks @2opremio!

**Pull request merged**